.github/workflows/net-perf-gke.yaml

```diff
@@ -0,0 +1,263 @@
+            hubble: "true"
+
+          - index: 6
+            name: "tunnel-hubble"
+            mode: "tunnel"
+            encryption: "none"
+            hubble: "true"
+
+          - index: 7
+            name: "native-ipsec-hubble"
+            mode: "native"
+            encryption: "ipsec"
+            hubble: "true"
+
+          - index: 8
+            name: "tunnel-ipsec-hubble"
+            mode: "tunnel"
+            encryption: "ipsec"
+            hubble: "true"
+
+    steps:
+      - name: Checkout context ref (trusted)
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          ref: ${{ inputs.context-ref || github.sha }}
+          persist-credentials: false
+
+      - name: Set Environment Variables
+        uses: ./.github/actions/set-env-variables
+
+      - name: Set up job variables
+        id: vars
+        run: |
+          if [ "${{ github.event_name }}" = "workflow_dispatch" ] || [ ${{ github.event.pull_request }} ] ; then
+            SHA="${{ inputs.image-tag }}"
+            OWNER="${{ inputs.PR-number }}"
+          else
+            SHA="${{ github.sha }}"
+            OWNER="${{ github.ref_name }}"
+            OWNER="${OWNER/./-}"
+          fi
+
+          CILIUM_INSTALL_DEFAULTS="--chart-directory=install/kubernetes/cilium \
+            --cluster-name=${{ env.clusterName }}-${{ matrix.index }} \
+            --helm-set=agentNotReadyTaintKey=ignore-taint.cluster-autoscaler.kubernetes.io/cilium-agent-not-ready \
+            --helm-set=debug.enabled=false \
+            --helm-set=bpf.monitorAggregation=maximum \
+            --helm-set=hubble.enabled=${{ matrix.hubble == 'true' }} \
+            --wait=false"
+
+          # only add SHA to the image tags if it was set
+          if [ -n "${SHA}" ]; then
+            echo sha=${SHA} >> $GITHUB_OUTPUT
+            CILIUM_INSTALL_DEFAULTS+=" --helm-set=image.repository=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/cilium-ci \
+            --helm-set=image.useDigest=false \
+            --helm-set=image.tag=${SHA} \
+            --helm-set=operator.image.repository=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/operator \
+            --helm-set=operator.image.suffix=-ci \
+            --helm-set=operator.image.tag=${SHA} \
+            --helm-set=operator.image.useDigest=false \
+            --helm-set=clustermesh.apiserver.image.repository=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/clustermesh-apiserver-ci \
+            --helm-set=clustermesh.apiserver.image.tag=${SHA} \
+            --helm-set=clustermesh.apiserver.image.useDigest=false \
+            --helm-set=hubble.relay.image.repository=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/hubble-relay-ci \
+            --helm-set=hubble.relay.image.tag=${SHA} \
+            --helm-set=hubble.relay.image.useDigest=false"
+          fi
+
+          if [ "${{ matrix.mode }}" = "tunnel" ] ; then
+            CILIUM_INSTALL_DEFAULTS+=" --datapath-mode=tunnel"
+          fi
```

Comment:
Nit. I'd explicitly configure the mode in both cases, instead of relying on the CLI default, as it could lead to surprises if ever using a different cloud provider.

Reply:
Makes sense, fixed. 

I'm not resolving this comment before rerunning tests to double check everything works.